### PR TITLE
273 migrate to terriajs

### PR DESCRIPTION
**Pull request by LukeParky**
DESCRIPTION OF PR:

Closes:
* #273 
* #162 
* #254 
* #255 
* #155 
* #148 
  (courtesy of PR #284)

Migrate to using TerriaJS as the primary front-end, enabling MEDUSA front end controls.
This requires the use of OGC WebProcessingService (WPS) to run processing tasks.

A new EnvVariable, `ROOF_SURFACE_DATASET_PATH` is added to `.env.template`. A new issue #283 will add this into git so that it doesn't need to be copied from network storage.

Another issue #280 is for removing the old front-end.

This PR reduces the size of the AOI, #282 will extend it again

## Developer Checklist
- [X] Make code change
- [ ] Update tests
  - [ ] Update / create new tests
  - [ ] Ensure these tests have the expected behaviour
  - [ ] Test locally and ensure tests are passing
- [X] Update documentation
  - [X] Readme
  - [X] Docstrings
  - [X] Comments
  - [ ] Wiki

## Reviewer Checklist
- [ ] Check new code for code smells
- [ ] Check new tests
  - [ ] Ensure adequate coverage
  - [ ] Check for code smells within tests
- [ ] Check if documentation needs updating
  - [ ] Readme
  - [ ] Docstrings
  - [ ] Comments
  - [ ] Wiki

### CODE REVIEW

File: src/pywps.cfg
Comment (LukeParky) [inline src/pywps.cfg:29]: Update this entire file

File: terriajs/client_config.json
Comment (LukeParky) [inline terriajs/client_config.json:8]: Use environment variables here (this key has already been disabled)

Comment (LukeParky) [reply terriajs/client_config.json:8]: Or remove the need for it

File: Dockerfile
Comment (LukeParky) [inline Dockerfile:1]: Update README with additional instructions for copying gdb file while waiting on approval from data owners

**Comment by LukeParky** [comment]
@caseyli92 this is essentially ready for a code review but additional documentation is required to enable you to easily test it. (you need the .gdb in the correct folder, will add more documentation around this shortly)

File: environment.yml
Comment (caseyli92) [inline environment.yml:9]: Could you please add a comment for this change?

File: src/app.py
Comment (caseyli92) [inline src/app.py:111]: The SonarCloud hotspot risk needs to be resolved.

File: src/digitaltwin/data_to_db.py
Comment (caseyli92) [inline src/digitaltwin/data_to_db.py:461]:
```suggestion
        log.info("Adding 'roof_surface_points' to the database.")
        roof_surface_points.to_postgis("roof_surface_points", engine, index=False, if_exists="replace")
        log.info("Successfully added 'roof_surface_points' to the database.")
```

File: src/digitaltwin/data_to_db.py
Comment (caseyli92) [inline src/digitaltwin/data_to_db.py:481]:
```suggestion
        log.info("Adding 'roof_surface_polygons' to the database.")
        roof_surface_polygons.to_postgis("roof_surface_polygons", engine, index=False, if_exists="replace")
        log.info("Successfully added 'roof_surface_polygons' to the database.")
```

File: src/digitaltwin/data_to_db.py
Comment (caseyli92) [inline src/digitaltwin/data_to_db.py:440]:
```suggestion
    """
    Read roof surface type points data then store them into database.
```

File: src/digitaltwin/data_to_db.py
Comment (caseyli92) [inline src/digitaltwin/data_to_db.py:465]:
```suggestion
    """
    Read roof surface polygons data then store them into database.
```

File: src/digitaltwin/data_to_db.py
Comment (caseyli92) [inline src/digitaltwin/data_to_db.py:449]:
```suggestion
    if check_table_exists(engine, "roof_surface_points"):
        log.info("'roof_surface_points' data already exists in the database.")
```

File: src/digitaltwin/data_to_db.py
Comment (caseyli92) [inline src/digitaltwin/data_to_db.py:474]:
```suggestion
    if check_table_exists(engine, "roof_surface_polygons"):
        log.info("'roof_surface_polygons' data already exists in the database.")
```

File: src/digitaltwin/data_to_db.py
Comment (caseyli92) [inline src/digitaltwin/data_to_db.py:487]:
```suggestion
    """
    Read roof surface type points and polygons from the data file and store them in the database.
```

File: src/digitaltwin/data_to_db.py
Comment (caseyli92) [inline src/digitaltwin/data_to_db.py:484]: avoid confusion

```suggestion
def save_roof_surface_data_to_db(engine: Engine) -> None:
```

File: src/digitaltwin/retrieve_static_boundaries.py
Comment (caseyli92) [inline src/digitaltwin/retrieve_static_boundaries.py:48]:
```suggestion
    data_to_db.save_roof_surface_data_to_db(engine)
```

File: src/pollution_model/run_medusa_2.py
Comment (LukeParky) [inline src/pollution_model/run_medusa_2.py:1]: Add descriptions for tests

**Review by caseyli92 — CHANGES_REQUESTED** [review]
Great work! I have a few suggestions and comments below. Also, please remember to add a short description to each of the Medusa unit tests.

File: Dockerfile
Comment (LukeParky) [reply Dockerfile:53]: The directory is created during copy by this command: https://github.com/GeospatialResearch/Digital-Twins/blob/aa6ba2363fa0e3eedb3e6acbf6ececd67c028e2b/Dockerfile#L52-L55 so the `roof_surfaces_data` directory does not need to be created.

The road surfaces aren't included in this PR so I will leave that for #286.

File: docker-compose-dev.yml
Comment (LukeParky) [reply docker-compose-dev.yml:24]: Deferring to #286, see comment https://github.com/GeospatialResearch/Digital-Twins/pull/281/files#r1845696839

**Review by caseyli92 — APPROVED** [review]

**Pull request merged**
